src/RelayClient.ts

```diff
@@ -134,6 +139,48 @@ class RelayClient extends EnvelopingEventEmitter {
 
     return true;
   }
+
+  public estimateInternalCallGas = async (
+    { request, relayData }: EnvelopingTypes.RelayRequestStruct,
+    internalEstimationCorrection: BigNumberish = INTERNAL_ESTIMATION_CORRECTION,
+    addCorrection = true
+  ): Promise<BigNumber> => {
+    let estimation = await this._provider.estimateGas({
+      from: request.from,
+      to: request.to,
+      gasPrice: relayData.gasPrice,
+      data: request.data,
+    });
+
+    // The INTERNAL_TRANSACTION_ESTIMATE_CORRECTION is substracted because the estimation is done using web3.eth.estimateGas which
+    // estimates the call as if it where an external call, and in our case it will be called internally (it's not the same cost).
+    // Because of this, the estimated maxPossibleGas in the server (which estimates the whole transaction) might not be enough to successfully pass
+    // the following verification made in the SmartWallet:
+    // require(gasleft() > req.gas, "Not enough gas left"). This is done right before calling the destination internally
+    if (estimation.gt(internalEstimationCorrection)) {
+      estimation = estimation.sub(internalEstimationCorrection);
+    }
```

Comment:
I think this part is shared with other [parts of the code](https://github.com/rsksmart/rif-relay-client/pull/60/files#diff-61c300ca57d8839a4c24a5d6700547d76cd72bca4c9ff8941705f5a7d4a0e983R207), so I'd include that logic (with the comment in an external function).